app/src/main/java/br/com/catbag/gifreduxsample/ui/components/GifComponent.java

```diff
@@ -102,29 +105,27 @@ public void setAnvilRenderListener(AnvilRenderListener listener) {
         mAnvilRenderListener = listener;
     }
 
+    @Override
+    public void onWindowFocusChanged(boolean hasWindowFocus) {
+        super.onWindowFocusChanged(hasWindowFocus);
+        if (hasWindowFocus) {
+            registerOnStateChange();
+        }
+        else {
+            unregisterOnStateChange();
+        }
+    }
```

Comment:
Duplicated code. Maybe we can extends RenderableView class to add this behavior. What do you think?